pages/docs/evolution/breaking-changes.md

```diff
@@ -0,0 +1,483 @@
+
+* `1.3.3` Backwards incompatible changes should be introduced through a deprecation when possible. 
+
+* `1.3.4` The tooling should provide means of detection and migration: users shouldn't be required to do much by hand.
+
+* `1.3.5` Users should be informed about such changes in a timely fashion (exact timeline is to be defined on a case-by-case basis). In particular, users should have enough time to migrate comfortably.
+
+`1.4` Any change that deprecates or breaks
+compatibility for a language feature or standard API needs to be reviewed by the [Kotlin
+Language Committee](language-committee.html) following the
+[review procedure](change-review-process.html).
+
+## `2` Changes to this policy
+
+`2.1` Changes to this policy need to be approved by
+    the Kotlin Language Committee.
+
+`2.2` Any proposed change needs to be published as
+    a GitHub pull request; providing a reasonable time to allow for
+     comments on the change by the Kotlin Community.
+
+## `3` Types of incompatible changes and corresponding guidelines
+
+`3.1` Small fixes that virtually no users will
+    encounter can normally be made right away (still require committee
+    [review](#1.4)).
+
+* `3.1.1` When unsure about the full risk/impact of
+    a change:
+
+*   Implement the change in a preview build.
+*   Try on large bodies of code available to us.
+*   Fall back to [deprecation procedure](#4) if needed.
+
+`3.2` Behavior-changing bug fixes in `kotlinc`:
+
+* `3.2.1` Good code did not compile due to a bug can be
+    fixed right away.
+
+* `3.2.2` Bad code compiled due to a bug, but always
+    failed at runtime can be fixed right away.
+
+* `3.2.3` Bad code compiled due to a bug, and worked
+    reasonably, must follow the deprecation policy spelled out in
+    [Paragraph 4](#4).
+
+
+`3.3` Bug fixes and contract refinement in the
+ `kotlin-stdlib` require publishing a release note one version in advance.
+
+`3.4` Retiring language features & `kotlin-stdlib` APIs
+    must follow the deprecation policies spelled out in
+    [Paragraph 4](#4) and [Paragraph 5](#5).
+
+
+
+## `4` Deprecation procedures for language features
+
+Deprecation is closely related to versioning, for versioning scheme
+we refer to the existing [versioning policy](../reference/compatibility.html).
+
+`4.1` Before any changes are made, the upcoming
+    change along with its rationale and the deprecation/migration plan
+    need to be announced to the Kotlin community, 
+    preview builds of the compiler and standard library should be made available.
+
+`4.2` Release a version (call it N.M) that reports
+    deprecation warnings for the feature:
+
+* `4.2.1` Provide and describe a replacement for
+    deprecated constructs as part of the warning.
+
+* `4.2.2` Allow to disable the warnings in settings,
+    or promote them to errors.
+
+* `4.2.3` Along with this version, provide automated
+    migration aids (e.g. in the IDE).
+
+`4.3` Allow a long enough period of time for
```

Comment:
IMHO it is important to specify in this document the exact minimum migration period.

Many teams wait a certain time before updating to a new version of a software to allow for the vendor to release "hot bug fix" updates. I assume those teams use the same approach to the Kotlin compiler, IDE and IDE plugin. 

The problem arises when a team's migration period is significantly bigger than Kotlin's migrating period. In that case it may happen that a team skips several Kotlin compiler version, thus not following the intended migration path.

By specifying the minimum migration period for Kotlin components, for example 45 days, the Kotlin team guaranties that anybody can update their toolchain once per 45 days and not miss a version.

To clarify, this would still allow the length of the migration period to vary "based on the impact and complexity of the change".

Reply:
Language versions don't come out every 45 days, it takes closer to a year to release a new version. These documents are intended to set gudielines, not provide guaratees, so there's no need to write out exact numbers.

Reply:
My comment related to minor language versions rather than to major versions. The minor versions come more often than once a year, but I agree with you that their cooking period (~ 6 months) is probably more than enough.

Reply:
From my perspective (university, teaching), it is important that I can continue using the same language version throughout the entire academic year while still having bug fixes for "significant" bugs (students are steered well away from edge cases and optimization is not an issue either). Changes in language versions throughout a course just increase opportunities for confusion as well as create problems with lab setup (which is often done once per year)

Reply:
Release versions are kept updated with important big fixes, and you don't have to migrate to a next release even if it's come out already, so I don't think there should be any problems in your case